src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -4468,6 +4573,945 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  void dilithium_load16zetas(int o0, Register zetas) {
+    __ ldpq(as_FloatRegister(o0), as_FloatRegister(o0 + 1), __ post (zetas, 32));
+    __ ldpq(as_FloatRegister(o0 + 2), as_FloatRegister(o0 + 3), __ post (zetas, 32));
+
+  }
+
+  void dilithium_load32zetas(Register zetas) {
+    dilithium_load16zetas(16, zetas);
+    dilithium_load16zetas(20, zetas);
+  }
+
+  // 2x16 32-bit Montgomery multiplications in parallel
+  // See the montMul() method of the sun.security.provider.ML_DSA class.
+  // Here MONT_R_BITS is 32, so the right shift by it is implicit.
+  // The constants MONT_Q_INV_MOD_R and MONT_Q are loaded in (all 32-bit
+  // chunks of) vector registers v30 and v31, resp.
+  // The inputs are in v0-v7 and v16-v23 and the results go to v16-v23,
+  // four 32-bit values in each register
```

Comment:
Suggestion:

Once again it would be good to annotate the lines in this code with comments  that relate the generated code back to the original Java code.

In the header comment you should refer to the relevant Java class and the var names there:

    // computes (in parallel across 8 x 4S vectors)
    //    a = b * c * 2^-32 mod MONT_Q
    // where
    //    inputs b and c are in v0, ..., v7 and v16, ... v23,
    //    scratch registers v24, ... v27 are clobbered
    //    output a is written back into v16, ... v23
    //    constants q and q_inv are in v30, v31
    //
    // See the equivalent Java code in method ML_DSA.montMul

Then comment the generation lines as shown below